proto/proto/store/rpc.proto

```diff
@@ -0,0 +1,319 @@
+    repeated primitives.SmtOpening proofs = 1;
+}
+
+// SYNC STATE
+// ================================================================================================
+
+// State synchronization request.
+//
+// Specifies state updates the client is interested in. The server will return the first block which
+// contains a note matching `note_tags` or the chain tip. And the corresponding updates to
+// `account_ids` for that block range.
+message SyncStateRequest {
+    // Last block known by the client. The response will contain data starting from the next block,
+    // until the first block which contains a note of matching the requested tag, or the chain tip
+    // if there are no notes.
+    fixed32 block_num = 1;
+
+    // Accounts' commitment to include in the response.
+    //
+    // An account commitment will be included if-and-only-if it is the latest update. Meaning it is
+    // possible there was an update to the account for the given range, but if it is not the latest,
+    // it won't be included in the response.
+    repeated account.AccountId account_ids = 2;
+
+    // Specifies the tags which the client is interested in.
+    repeated fixed32 note_tags = 3;
+}
+
+// Represents the result of syncing state request.
+message SyncStateResponse {
+    // Number of the latest block in the chain.
+    fixed32 chain_tip = 1;
+
+    // Block header of the block with the first note matching the specified criteria.
+    blockchain.BlockHeader block_header = 2;
+
+    // Data needed to update the partial MMR from `request.block_num + 1` to `response.block_header.block_num`.
+    primitives.MmrDelta mmr_delta = 3;
+
+    // List of account commitments updated after `request.block_num + 1` but not after `response.block_header.block_num`.
+    repeated account.AccountSummary accounts = 5;
+
+    // List of transactions executed against requested accounts between `request.block_num + 1` and
+    // `response.block_header.block_num`.
+    repeated transaction.TransactionSummary transactions = 6;
+
+    // List of all notes together with the Merkle paths from `response.block_header.note_root`.
+    repeated note.NoteSyncRecord notes = 7;
+}
+
+// SYNC NOTES
+// ================================================================================================
+
+// Note synchronization request.
+//
+// Specifies note tags that client is interested in. The server will return the first block which
+// contains a note matching `note_tags` or the chain tip.
+message SyncNotesRequest {
+    // Last block known by the client. The response will contain data starting from the next block,
+    // until the first block which contains a note of matching the requested tag.
+    fixed32 block_num = 1;
+
+    // Specifies the tags which the client is interested in.
+    repeated fixed32 note_tags = 2;
+}
+
+// Represents the result of syncing notes request.
+message SyncNotesResponse {
```

Comment:
Part of me wants to strive for fully replacing the request/response suffixes. Mainly because that naming tends to be less expressive than the alternative. Understood that sometimes it may be very difficult to think of an appropriate / expressive name in the context.

Taking this as an example. What would we rename `SyncNotesResponse` to? Perhaps `NoteSyncRecordsByTag` or `..ByTagAndBlock` which gets verbose but you get the idea.

LMK what you think. cc @bobbinth